integrated_channels/blackboard/client.py

```diff
@@ -40,15 +45,235 @@ def __init__(self, enterprise_configuration):
+            raise ClientError("No externalId found in metadata, please check json data format", 400)
+
+        external_id = channel_metadata_item.get('externalId')
+
+        # blackboard does not support all characters in our courseIds so let's gen a hash instead
+        course_id_generated = abs(hash(external_id))
+
+        copy_of_channel_metadata = copy.deepcopy(channel_metadata_item)
+        copy_of_channel_metadata['courseId'] = course_id_generated
+
+        serialized_channel_metadata = json.dumps(copy_of_channel_metadata).encode('utf-8')
+
+        LOGGER.info("Creating course with courseId: %s", external_id)
+        self._create_session()
+        response = self._post(self.create_course_url, serialized_channel_metadata)
+        return response.status_code, response.text
 
     def update_content_metadata(self, serialized_data):
-        """TODO"""
+        """Apply changes to a course if applicable"""
+        self._create_session()
+        channel_metadata_item = json.loads(serialized_data.decode("utf-8"))
+
+        BlackboardAPIClient._raise_for_external_id(channel_metadata_item)
+        external_id = channel_metadata_item.get('externalId')
+        course_id = self._resolve_blackboard_course_id(external_id)
+        BlackboardAPIClient._raise_for_course_id(course_id, external_id)
+
+        LOGGER.info("Updating course with courseId: %s", course_id)
+        response = self._patch(COURSE_V3_PATH.format(course_id=course_id), serialized_data)
+        return response.status_code, response.text
 
     def delete_content_metadata(self, serialized_data):
-        """TODO"""
+        """Delete course from blackboard (performs full delete as of now)"""
+        self._create_session()
+        channel_metadata_item = json.loads(serialized_data.decode("utf-8"))
+
+        BlackboardAPIClient._raise_for_external_id(channel_metadata_item)
+        external_id = channel_metadata_item.get('externalId')
+        course_id = self._resolve_blackboard_course_id(external_id)
+        BlackboardAPIClient._raise_for_course_id(course_id, external_id)
+
+        LOGGER.info("Deleting course with courseId: %s", course_id)
+        url = COURSE_V3_PATH.format(course_id=course_id)
+        response = self._delete(url)
+        return response.status_code, response.text
+
+    def create_course_completion(self, user_id, payload):
+        """
+        Post a final course grade to the integrated Blackboard course.
+
+        Parameters:
+        -----------
+            user_id (str): The shared email between a user's edX account and Blackboard account
+            payload (str): The (string representation) of the learner data information
+
+        Example payload:
+        ---------------
+            '{
+                courseID: course-edx+555+3T2020,
+                score: 0.85,
+                completedTimestamp: 1602265162589,
+            }'
+
+        """
+        self._create_session()
+        learner_data = json.loads(payload)
```

Comment:
renamed a couple vars in this function, fyi, for clarity

Reply:
that's fine with me :)